server.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+)
+
+func fallback(w http.ResponseWriter, r *http.Request, reason string) {
+	location := "http://redirect.name/"
+	if reason != "" {
+		location = fmt.Sprintf("%s#reason=%s", location, url.QueryEscape(reason))
+	}
+	http.Redirect(w, r, location, 302)
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	host, _, err := net.SplitHostPort(r.Host)
+	if err != nil {
+		fallback(w, r, fmt.Sprintf("Could not split host (%v)", err))
+		return
+	}
+
+	hostname := fmt.Sprintf("_redirect.%s", host)
+	txt, err := net.LookupTXT(hostname)
```

Comment:
If you find that the built-in DNS resolver doesn't provide enough control, consider https://github.com/miekg/dns